docs/userguide/garbage_collection.rst

```diff
@@ -0,0 +1,33 @@
+.. _garbage_collection:
+
+Garbage Collection
+------------------
+
+Production workflows often run for days processing millions of tasks, and as tasks accumulate keeping
```

Comment:
I dislike this paragraph. It comes from the angle of "this is how parsl used to work internally and this is how it has been changed" - that belongs in the changelog / PR commit message.

There's no change to the user exposed interface by garbage collection, and so no need to change the documentation.

With memoization/checkpointing on or off, the memory usage of parsl previously differed. Now that difference is more. We didn't document that before. If we're going to document that now, document it as an improvement to the checkpointing documentation.

Reply:
I'm not sure how to improve this. I tried to write this as a feature note, so that users know this exists, maybe this doesn't warrant any attention.

I don't mind removing this entirely and just leaving a note in checkpointing that some modes can interfere with garbage collection.

Reply:
I vote to remove this and leave a note in checkpointing

Reply:
(i agree with @benclifford)

Reply:
Done.